### PR TITLE
moves api and exposed default api impl into new packages

### PR DESCRIPTION
**Pull request by peterhuffer**
#### What does this PR do?
- bundle exports both these packages and hides the default processing framework and the post ingest plugin

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@emmberk @troymohl 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@emmberk 

#### How should this be tested? (List steps with links to updated documentation)
full build against empty repo.
ingest nitf and ensure thumbnail, overview, and original images all available

#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/plugin/catalog-plugin-async/src/main/java/org/codice/ddf/catalog/plugin/async/api/internal/ProcessRequest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.List;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface ProcessRequest {
+    // TODO implement Operation.java
```
Comment (troymohl) [inline catalog/plugin/catalog-plugin-async/src/main/java/org/codice/ddf/catalog/plugin/async/api/internal/ProcessRequest.java]: Is this still a TODO

Comment (peterhuffer) [inline catalog/plugin/catalog-plugin-async/src/main/java/org/codice/ddf/catalog/plugin/async/api/internal/ProcessRequest.java]: Yeah. @emmberk will be handling this I believe.

**Comment by emmberk** [comment]
Working on hero now

**Comment by emmberk** [comment]
Should the `DDF :: Catalog :: Plugin` pom say `<module>async</module>` instead of `<module>catalog-plugin-async</module>`

**Pull request merged**